osu.Game.Rulesets.Osu/Difficulty/Evaluators/SpeedEvaluator.cs

```diff
@@ -56,10 +56,28 @@ public static double EvaluateDifficultyOf(DifficultyHitObject current)
             if (strainTime < min_speed_bonus)
                 speedBonus = 1 + 0.75 * Math.Pow((min_speed_bonus - strainTime) / speed_balancing_factor, 2);
 
-            double travelDistance = osuPrevObj?.TravelDistance ?? 0;
-            double distance = Math.Min(single_spacing_threshold, travelDistance + osuCurrObj.MinimumJumpDistance);
+            double distance = Math.Min(single_spacing_threshold, osuCurrObj.JumpDistance);
 
-            return (speedBonus + speedBonus * Math.Pow(distance / single_spacing_threshold, 3.5)) * doubletapness / strainTime;
+            double sliderStreamMultiplier = 1;
+
+            if (osuCurrObj.BaseObject is Slider slider && osuPrevObj?.BaseObject is Slider)
+            {
+                // Bonus base
+                double sliderStreamBonus = 0.25;
+
+                // If slider was slower than notes before - punish it
+                if (osuCurrObj.StrainTime > osuPrevObj.StrainTime)
+                    sliderStreamBonus *= Math.Pow(Math.Min(osuCurrObj.StrainTime, osuPrevObj.StrainTime) / Math.Max(osuCurrObj.StrainTime, osuPrevObj.StrainTime), 2);
+
+                // Punish too short sliders to prevent cheesing (cheesing is still possible, but it's very rare)
+                double sliderLength = slider.Velocity * slider.SpanDuration;
+                if (sliderLength < slider.Radius / 2)
+                    sliderStreamBonus *= sliderLength / (slider.Radius / 2);
```

Comment:
woah -- this is different to the `AimEvaluator` case thanks to the divisions by 2. maybe needs a comment to explain in case this is intentional

Reply:
it's intentional cuz sliderstreams usually have shorter sliders than sliderjumps, and i don't want to be as harsh in anti-cheese here

Reply:
that doesn't make any sense. so if sliderjump cheese length sliders were used for sliderstreams, they suddenly wouldn't be worthy of receiving cheesing penalty?

Reply:
okay, i will change this